rust/chains/hyperlane-cosmos/src/interchain_security_module.rs

```diff
@@ -91,6 +93,23 @@ impl InterchainSecurityModule for CosmosInterchainSecurityModule {
         message: &HyperlaneMessage,
         metadata: &[u8],
     ) -> ChainResult<Option<U256>> {
-        Ok(Some(U256::from(1000))) // TODO
+        let payload = VerifyRequest {
+            verify: VerifyRequestInner {
+                metadata: hex::encode(metadata),
+                message: hex::encode(RawHyperlaneMessage::from(message)),
+            },
+        };
+        let data = self
+            .provider
+            .grpc()
+            .wasm_query(QueryIsmGeneralRequest { ism: payload }, None)
+            .await?;
+        let response: VerifyResponse = serde_json::from_slice(&data)?;
+        // We can't simulate the `verify` call in CosmWasm because
+        // it's not marked as an entrypoint. So we just use the query interface
+        // and hardcode a gas value - this can be inefficient if one ISM is
+        // vastly cheaper than another one.
+        let dummy_gas_value = U256::one();
```

Comment:
This is unfortunate :( Did you see if Mitosis had any ideas on what to do here? I guess we can leave this as-is and create and issue to track that this is incomplete.

If we really need this to be accurate in the future we can consider creating a contract that calls into the ISMs to verify

Also slight nit for the comment for posterity, `verify` is accessible via the query entrypoint but just not the execute entrypoint - when I first read this I was surprised that verify wasn't accessible via an entrypoint because it is able to be called by other contracts on chain https://book.cosmwasm.com/basics/entry-points.html

Reply:
> verify is accessible via the query entrypoint
That's what we're using, I don't think we can use it to simulate txs though?

Will check with Mitosis as well